Userland/Libraries/LibWeb/URL/URLSearchParamsIterator.cpp

```diff
@@ -42,9 +42,26 @@ JS::ThrowCompletionOr<void> URLSearchParamsIterator::initialize(JS::Realm& realm
     MUST_OR_THROW_OOM(Base::initialize(realm));
     set_prototype(&Bindings::ensure_web_prototype<Bindings::URLSearchParamsIteratorPrototype>(realm, "URLSearchParamsIterator"));
 
+    auto& vm = realm.vm();
+    u8 attr = JS::Attribute::Writable | JS::Attribute::Configurable;
+    define_native_function(realm, vm.well_known_symbol_iterator(), symbol_iterator, 0, attr);
+
     return {};
 }
 
+static JS::ThrowCompletionOr<URLSearchParamsIterator*> impl_from(JS::VM& vm)
+{
+    auto this_object = TRY(vm.this_value().to_object(vm));
+    if (!is<URLSearchParamsIterator>(*this_object))
+        return vm.throw_completion<JS::TypeError>(JS::ErrorType::NotAnObjectOfType, "@name@");
+    return static_cast<URLSearchParamsIterator*>(this_object.ptr());
+}
+
+JS_DEFINE_NATIVE_FUNCTION(URLSearchParamsIterator::symbol_iterator)
+{
+    return TRY(impl_from(vm));
```

Comment:
This doesn't match the major 3 which just return the this value with no validation:
Firefox:
![image](https://github.com/SerenityOS/serenity/assets/25595356/a919f92e-8526-4230-9efd-278560fad572)

Chrome:
![image](https://github.com/SerenityOS/serenity/assets/25595356/0378d475-d4ff-42df-bf2c-de3336528768)

Epiphany (WebKit):
![image](https://github.com/SerenityOS/serenity/assets/25595356/4d2b79d4-4ffd-4d3e-959b-7daf659f7494)

Reply:
Thanks for the fact-check!! That makes it much easier :)